framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -169,12 +169,9 @@ IterationAdaptiveDT::init()
         const auto * plfunc = dynamic_cast<const PiecewiseLinear *>(pfunc);
         _piecewise_linear_timestep_limiting_functions.push_back(plfunc);
 
-        if (plfunc)
```

Comment:
@dschwen could it be this easy?!

Reply:
uhm... looks like all you do is removing the nullptr check for the value returned from the dynamic cast. This code would most likely crash for all functions that are not derived from (or are of type) `PiecewiseLinear`

Reply:
No, I also changed it from `plfunc->domain` to `pfunc->domain. It's protected in the loop above

Reply:
change `PiecewiseLinear` to `PiecewiseBase` and put the check back in. that might do what you want!

Reply:
both `functionSize` and `domain` are declared in `PiecewiseBase`

Reply:
I don't think so...read comment above.

Reply:
I'm sampling from `pfunc` which is `PiecewiseBase`

Reply:
Ah, right, I see now. Then yes, if all you want is to force steps at every function point, that would be it. Is that where the left,right,centerdd comes in, because a value of a piecewise constant function is not really well defined at function points, is it?

Reply:
It's independent of that other PR, which fixed a massive frustration I had. There is always a discontinuity, but I don't think that comes into play here, nor there.

Reply:
...unless I'm missing something?

Reply:
This PR adds a time point at a function node of a piecewise constant function. That point is right at the discontinuity, isn't it? So I think it should at least be documented for the user which function value is chosen at those points.

Reply:
PiecewiseConstant chooses what to do at that discontinuity, that's the whole reason for all those switches, directions, etc.